src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -953,29 +955,35 @@ private boolean checkForSaveAndContinue() {
     return willContinue;
   }
 
-  private boolean checkForSaveAndExecute() {
-    boolean willContinue = false;
+  private String commitConfigToFileAndGetFilepath() {
+    String filename = null;
     if (checkIfNeedsSaving()) {
       ButtonType saveButton = new ButtonType("Save", ButtonBar.ButtonData.YES);
+      ButtonType dontSaveButton = new ButtonType("Proceed without Saving", ButtonBar.ButtonData.NO);
```

Comment:
1) We need a cancel button too.
2) If a user clicks the X in the top right corner of the dialog, that should also act as cancel (it currently proceeds with the tabulation).

Reply:
Three buttons seems pretty confusing (and is [poor UX practice](https://ux.stackexchange.com/a/131474) if it can be avoided).

If the user hits tabulate and there are no changes, there is no opportunity to cancel. Why would it be different if there are changes? Just in case the changes were unintended?

I'm okay hooking up the `X` to cancel` but hoping we can avoid a three-button dialog.

Reply:
There existing dialog warning users that they need to save it has a cancel button; and yeah, in case the changes were unintended is a good reason.

Sure, I guess I'm cool with still allowing the user to cancel by pressing the X if you don't want to add a third button.

Reply:
The X button is still proceeding with the tabulation.

Reply:
BTW, it appears we do have existing dialogs that have three buttons; see `checkForSaveAndContinue`.

Reply:
Fair enough -- I'll have three buttons only for developers, which makes me feel better about it. The third button only appears if version is TEST.